sktime/param_est/seasonality.py

```diff
@@ -407,3 +406,93 @@ def get_test_params(cls, parameter_set="default"):
+
+
+class Seasonality(BaseParamFitter):
+    """Find candidate seasonality parameter.
+
+    Interfacing `seasonal.periodogram` to determine candidate seasonality parameters.
+
+    Attributes
+    ----------
+    sp_ : int, seasonality period at lowest p-level, if any sub-threshold, else 1
+        if `candidate_sp` is passed, will be in `candidate_sp` or 1
+    sp_significant_ : list of int, seasonality periods with sub-threshold p-levels
+        ordered increasingly by p-level. Empty list, not [1], if none are sub-threshold
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.param_est.seasonality import Seasonality
+    >>>
+    >>> X = load_airline().diff()[1:]  # doctest: +SKIP
+    >>> sp_est = Seasonality()  # doctest: +SKIP
+    >>> sp_est.fit(X)  # doctest: +SKIP
+    Seasonality(...)
+    >>> sp_est.get_fitted_params()["sp"]  # doctest: +SKIP
+    12
+    >>> sp_est.get_fitted_params()["sp_significant"]  # doctest: +SKIP
+    array([12, 11])
+    """
+
+    _tags = {
+        "X_inner_mtype": "pd.Series",
+        "scitype:X": "Series",
+        "capability:missing_values": True,
+        "capability:multivariate": False,
+        "python_dependencies": "seasonal",
+    }
+
+    def __init__(self):
+        super(Seasonality, self).__init__()
+
+    def _fit(self, X):
+        """Fit estimator and estimate parameters.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            Time series to which to fit the estimator.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        seasons, _ = periodogram(X)
+        if seasons is None:
+            self.sp_ = 1
+            self.sp_significant_ = []
+        else:
+            self.sp_significant_ = seasons
+            self.sp_ = self.sp_significant_[0]
+
+        return self
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for transformers.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        params1 = {}
```

Comment:
Why leave it empty? Unless I'm misunderstanding something, the entire purpose of this method is to overwrite default params to facilitate faster testing. I suggest making use of a set of params that maximizes code coverage.

Reply:
I see I made a few mistakes here

1. There are no parameters required for this class, so we don't need them anyways
2. I have not written any tests for it either (simple interfacing task)

(Copied Franz Kiraly's code for reference, but forgot to delete this part)

Reply:
it has `min_period` and `max_period`, I would expose these as parameters, possibly through `candidate_sp`

Reply:
Done! I also wrote test parameters for it, will write tests for it later